dask/array/tests/test_overlap.py

```diff
@@ -826,3 +826,9 @@ def test_sliding_window_errors(window_shape, axis):
     arr = da.zeros((4, 3))
     with pytest.raises(ValueError):
         sliding_window_view(arr, window_shape, axis)
+
+
+def test_map_overlap_new_axis():
+    arr = da.arange(6, chunks=2)
+    actual = arr.map_overlap(lambda x: np.stack([x, x + 0.5]), depth=1, new_axis=[0])
+    assert_eq(np.stack([np.arange(6), np.arange(6) + 0.5]), actual.compute())
```

Comment:
Can you remove the compute here? assert_eq will do this automatically

Can you also assert the chunks of actual to make sure that things work as expected?

Reply:
I added a check for the output shape.

Without `compute()`, the assert_eq call fails:

```python
>       assert_eq(expected, actual)

dask/array/tests/test_overlap.py:839: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dask/array/utils.py:324: in assert_eq
    b, bdt, b_meta, b_computed = _get_dt_meta_computed(
dask/array/utils.py:275: in _get_dt_meta_computed
    x = _check_chunks(x, check_ndim=check_ndim, scheduler=scheduler)
dask/array/utils.py:246: in _check_chunks
    assert_eq_shape(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = (1, 2), b = (2, 2), check_ndim = True, check_nan = False

    def assert_eq_shape(a, b, check_ndim=True, check_nan=True):
        if check_ndim:
            assert len(a) == len(b)
    
        for aa, bb in zip(a, b):
            if math.isnan(aa) or math.isnan(bb):
                if check_nan:
                    assert math.isnan(aa) == math.isnan(bb)
            else:
>               assert aa == bb
E               assert 1 == 2

dask/array/utils.py:234: AssertionError
```

Reply:
Thx.

The fact that this fails means that your chunk sizes in the output are wrong (or another attribute that defines the shape)

Reply:
I *think* this has revealed a different bug with how `map_blocks` sets the shape of its output: https://github.com/dask/dask/issues/11329. Will convert this issue to draft until that one is solved.

Reply:
Thanks for digging in here. I think we can move ahead here since this is a different issue and not a result of your PR.

Reply:
Just add comment in the test pointing to the issue you created

Re the test: Please pass the uncomputed result in and set check_shape=False, this communicates better what the issue is

Reply:
👍 done, added a comment, and had to specify check_chunks=False too.